src/bin/hal-elements/cmd/mod.rs

```diff
@@ -95,7 +95,7 @@ pub fn arg_or_stdin<'a>(matches: &'a clap::ArgMatches<'a>, arg: &str) -> Cow<'a,
 		if input.is_empty() {
 			panic!("no '{}' argument given", arg);
 		}
-		String::from_utf8(input).expect(&format!("invalid utf8 on stdin for '{}'", arg))
```

Comment:
May as well include the error itself in the message (on the off-chance that it's useful to someone).

```
String::from_utf8(input).unwrap_or_else(|err| panic!("invalid utf8 on stdin for '{}': {}", arg, err))
```

Also that while loop a few lines up looks pointless(?) Is there a story behind it?

Reply:
Updated to add the extra error.

As for the loop, I have no idea. The earliest instance of it appears in hal on April 22. There is no associated PR, just a direct push to master by Steven, and no explanation. You can take a look at 68b9d5d7d3b929610a13c16cf0915474c2fdd1ae if you want.